### ISSUE TITLE
Winml perf use DML by default when device set to gpu

### ISSUE THREAD

**Issue by ssss141414** [microsoft/winml-cli#649]
(Downloads) PS C:\Users\shzhen\Downloads> winml perf -m convnext/model_opt.onnx --device gpu --iterations 100
Benchmarking ONNX: convnext\model_opt.onnx

Device:      gpu / DmlExecutionProvider
Inputs:      pixel_values         [1, 3, 224, 224]       float32
Outputs:     logits               [1, 1000]

Device:      gpu / DmlExecutionProvider
Task:        n/a (direct ONNX)

Expect: check and use GPU like Nvidia GPU first:

**Comment by DingmaomaoBJTU** [comment]
**Triage summary**

- **Component**: Perf
- **Priority**: P1

When `--device gpu` is specified, `winml perf` always defaults to `DmlExecutionProvider` instead of checking for a higher-priority GPU EP (e.g., CUDA for Nvidia GPUs) first. The EP selection logic should prioritize native GPU EPs before falling back to DML.

**Comment by xieofxie** [comment]
resolved in 0.0.3 fix
